elements/pfe-navigation/src/pfe-navigation-item.js

```diff
@@ -0,0 +1,429 @@
+        this.navigationWrapper._activeNavigationItems.push(this);
+  
+        this.expanded = true;
+        this.navigationWrapper.overlay = true;
+      }
+  
+      // Dispatch the event
+      this.dispatchEvent(
+        new CustomEvent(`${this.tag}:open`, {
+          detail: {},
+          bubbles: true,
+          composed: true
+        })
+      );
+    }
+  
+    close(event) {
+      if (event) event.preventDefault();
+  
+      // Close the children elements
+      this.navigationWrapper._activeNavigationItems = this.navigationWrapper._activeNavigationItems.filter(item => {
+        let close = this.nestedItems && this.nestedItems.includes(item);
+        if (close) item.close();
+        return !close && item !== this;
+      });
+  
+      this.expanded = false;
+  
+      // Clear the overlay
+      this.navigationWrapper.overlay = this.navigationWrapper._activeNavigationItems.length > 0;
+  
+      this.focus();
+  
+      // Dispatch the event
+      this.dispatchEvent(
+        new CustomEvent(`${this.tag}:close`, {
+          detail: {},
+          bubbles: true,
+          composed: true
+        })
+      );
+    }
+  
+    toggle(event) {
+      if (event) event.preventDefault();
+  
+      if (this.visible && !this.expanded) {
+        this.open(event);
+      } else {
+        this.close(event);
+      }
+    }
+  
+    constructor() {
+      super(PfeNavigationItem);
+  
+      // States
+      this.nested = false;
+      this.expanded = false;
+  
+      // Objects
+      this.trigger = null;
+      this.tray = null;
+      this.directLink = null;
+      this.linkUrl = null;
+  
+      // Lists
+      this.nestedItems = [];
+  
+      // Shadow elements
+      this._trigger = this.shadowRoot.querySelector(`.${this.tag}__trigger`);
+      this._icon = this.shadowRoot.querySelector("pfe-icon");
+      this._tray = this.shadowRoot.querySelector(`.${this.tag}__tray`);
+  
+      // Externally accessible events
+      this.close = this.close.bind(this);
+      this.open = this.open.bind(this);
+      this.toggle = this.toggle.bind(this);
+  
+      // Initializers
+      this._init__trigger = this._init__trigger.bind(this);
+      this._init__tray = this._init__tray.bind(this);
+  
+      // Event handlers
+      this._keydownHandler = this._keydownHandler.bind(this);
+      this._keyupHandler = this._keyupHandler.bind(this);
+      this._navigateToUrl = this._navigateToUrl.bind(this);
+      this._directLinkHandler = this._directLinkHandler.bind(this);
+    }
+  
+    connectedCallback() {
+      super.connectedCallback();
+  
+      this._init__trigger();
+      this._init__tray();
+  
+      // Add a slotchange listeners to the lightDOM elements
+      if (this.trigger) this.trigger.addEventListener("slotchange", this._init__trigger);
+      if (this.tray) this.tray.addEventListener("slotchange", this._init__tray);
+    }
+  
+    attributeChangedCallback(attr, oldValue, newValue) {
```

Comment:
Is this necessary if we're not doing anything inside the `attributeChangedCallback`?

Reply:
Probably not - I'll remove.